bpf/tc/attach.go

```diff
@@ -100,20 +100,57 @@ func (ap AttachPoint) AttachProgram() error {
 	defer tcLock.RUnlock()
 	logCxt.Debug("AttachProgram got lock.")
 
+	//nolint
 	progsToClean, err := ap.listAttachedPrograms()
 	if err != nil {
-		return err
+		return nil, err
+	}
+	obj, err := libbpf.OpenObject(tempBinary)
+	if err != nil {
+		return nil, err
+	}
+	defer obj.Close()
+
+	baseDir := "/sys/fs/bpf/tc/"
+	for m, err := obj.FirstMap(); m != nil && err == nil; m, err = m.NextMap() {
+		subDir := "globals"
+		if m.Type() == libbpf.MapTypeProgrArray && strings.Contains(m.Name(), "cali_jump") {
+			if ap.Hook == HookIngress {
+				subDir = ap.Iface + "_igr/"
+			} else {
+				subDir = ap.Iface + "_egr/"
+			}
+		}
+		pinPath := path.Join(baseDir, subDir, m.Name())
+		perr := m.SetPinPath(pinPath)
+		if perr != nil {
+			return nil, fmt.Errorf("error pinning map %v errno %v", m.Name(), perr)
+		}
 	}
 
-	_, err = ExecTC("filter", "add", "dev", ap.Iface, string(ap.Hook),
-		"bpf", "da", "obj", tempBinary,
-		"sec", SectionName(ap.Type, ap.ToOrFrom),
-	)
+	err = obj.Load()
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("error loading program %v", err)
+	}
+
+	isHost := false
+	if ap.Type == "host" {
+		isHost = true
+	}
+
+	err = updateJumpMap(obj, isHost)
```

Comment:
Since we can modify the map before attaching, would be great to install the right policy program at this stage rather than a temp one.  Feels like a follow-on piece of work though.